albatross/covariance_functions/distance_metrics.h

```diff
@@ -45,17 +45,38 @@ class EuclideanDistance : public DistanceMetric {
   }
 };
 
+template <typename _Scalar, int _Rows>
+double radial_distance(const Eigen::Matrix<_Scalar, _Rows, 1> &x,
+                       const Eigen::Matrix<_Scalar, _Rows, 1> &y) {
+  return fabs(x.norm() - y.norm());
+}
+
 class RadialDistance : public DistanceMetric {
 public:
   ~RadialDistance(){};
 
   std::string get_name() const override { return "radial_distance"; };
 
   double operator()(const Eigen::VectorXd &x, const Eigen::VectorXd &y) const {
-    return fabs(x.norm() - y.norm());
+    return radial_distance(x, y);
   }
 };
 
+template <typename _Scalar, int _Rows>
+double angular_distance(const Eigen::Matrix<_Scalar, _Rows, 1> &x,
+                        const Eigen::Matrix<_Scalar, _Rows, 1> &y) {
+  // The acos operator doesn't behave well near |1|.  acos(1.), for example,
+  // returns NaN, so here we do some special casing,
+  double dot_product = x.dot(y) / (x.norm() * y.norm());
+  if (dot_product > 1. - 1e-16) {
```

Comment:
Can we name `1e-16` as "epsilon" somewhere so we can reuse it?

Reply:
done